runtime/jvmti/jvmtiStartup.c

```diff
@@ -379,31 +379,32 @@ issueAgentOnLoadAttach(J9JavaVM * vm, J9JVMTIAgentLibrary * agentLibrary, const
 		/* agentInitFunction invokes java.instrument/share/native/libinstrument/InvocationAdapter.c
 		 * which expects jarPath in system default code page encoding.
 		 */
-		char *tempJarPath = NULL;
-		BOOLEAN conversionSucceed = FALSE;
-		int32_t size = j9str_convert(J9STR_CODE_MUTF8, J9STR_CODE_WINDEFAULTACP, options, strlen(options), NULL, 0);
-		if (size > 0) {
-			size += 1; /* leave room for null */
-			tempJarPath = j9mem_allocate_memory(size, OMRMEM_CATEGORY_VM);
-			if (NULL != tempJarPath) {
-				size = j9str_convert(J9STR_CODE_MUTF8, J9STR_CODE_WINDEFAULTACP, options, strlen(options), tempJarPath, size);
-				if (size > 0) {
-					conversionSucceed = TRUE;
+		IDATA optionLen = strlen(options);
```

Comment:
Is there a requirement that options is not null?  Previously the code wasn't handling that case either

Reply:
As per `JVMTI` spec [1], _If startup options were not provided, a zero length string is passed to options_. So it can't be `null`.
Looked the callers of `issueAgentOnLoadAttach()`, it was assigned to a non-null pointer.
https://github.com/eclipse/openj9/blob/ecadae97e4d40cf6ec548b4fe7f2795738a740b9/runtime/jvmti/jvmtiStartup.c#L979-L983
https://github.com/eclipse/openj9/blob/ecadae97e4d40cf6ec548b4fe7f2795738a740b9/runtime/vm/BytecodeInterpreter.hpp#L4367-L4369


[1] https://docs.oracle.com/javase/8/docs/platform/jvmti/jvmti.html#onattach